app.py

```diff
@@ -68,12 +94,182 @@
     _LOGGER.info("installation=%s", RUNTIME_CONTEXT.app_installation)
 
 
+@process_event_actions("pull_request", {"opened", "reopened", "synchronize", "edited"})
+@process_webhook_payload
+async def on_pr_open_or_sync(*, action, number, pull_request, repository, sender, installation, **kwargs):
+    """React to an opened or changed PR event.
+
+    Send a status update to GitHub via Checks API.
+    """
+    _LOGGER.info(
+        f"on_pr_open_or_sync: working on PR {pull_request['html_url']}",
+    )
+
+    github_api = RUNTIME_CONTEXT.app_installation_client
+
+    pr_head_sha = pull_request["head"]["sha"]
+    repo_url = pull_request["base"]["repo"]["url"]
+    check_runs_base_uri = f"{repo_url}/check-runs"
+
+    if pr_head_sha is None:
+        _LOGGER.error(f"on_pr_open_or_sync: no Pull Request head sha found, stopped working!")
+        return
+
+    _LOGGER.info(f"on_pr_open_or_sync: PR commit id {pr_head_sha} will be used for check-run")
+
+    resp = await github_api.post(
+        check_runs_base_uri,
+        preview_api_version="antiope",
+        data={
+            "name": CHECK_RUN_NAME,
+            "head_sha": pr_head_sha,
+            "status": "queued",
+            "started_at": f"{datetime.utcnow().isoformat()}Z",
+        },
+    )
+
+    check_run_id = int(resp["id"])  # TODO do we need some marshaling here?
+
+    check_runs_updates_uri = f"{check_runs_base_uri}/{check_run_id}"
+    _LOGGER.info(f"on_pr_open_or_sync: check_run_id: {check_run_id}")
+
+    _submit_thamos_workflow(
+        check_run_id,
+        repo_url=repo_url,
+        revision=pr_head_sha,
+        installation=installation
+    )
+
+    resp = await github_api.patch(
+        check_runs_updates_uri, preview_api_version="antiope", data={"name": CHECK_RUN_NAME, "status": "in_progress"},
+    )
+
+    _LOGGER.info(
+        f"on_pr_open_or_sync: working on PR %s: sleeping a random time between 5 and 15 seconds...",
+        pull_request["html_url"],
+    )
+
+    timeDelay = random.randrange(5, 15)
+    time.sleep(timeDelay)
```

Comment:
Using this in the async code is a suicide. It stops your whole app.

you must use `asyncio.sleep` instead or a helper `github_api.sleep`